Tests/ttx/ttx_test.py

```diff
@@ -263,5 +267,569 @@ def test_guessFileType_invalid_path(self):
+    options = ttx.Options([("", "")], 1)
+    options.listTables = True
+    ttx.ttList(inpath, fakeoutpath, options)
+    out, err = capsys.readouterr()
+    expected_tables = ("head", "hhea", "maxp", "OS/2", "name", "cmap", "hmtx", "fpgm", "prep", "cvt ", "loca", "glyf", "post", "gasp", "DSIG")
+    # confirm that expected tables are printed to stdout
+    for table in expected_tables:
+        assert table in out
+    # test for one of the expected tag/checksum/length/offset strings
+    assert "OS/2  0x67230FF8        96       376" in out
+
+
+def test_ttx_ttlist_otf(capsys, tmpdir):
+    inpath = os.path.join("Tests", "ttx", "data", "TestOTF.otf")
+    fakeoutpath = os.path.join(str(tmpdir), "TestOTF.ttx")
+    options = ttx.Options([("", "")], 1)
+    options.listTables = True
+    ttx.ttList(inpath, fakeoutpath, options)
+    out, err = capsys.readouterr()
+    expected_tables = ("head", "hhea", "maxp", "OS/2", "name", "cmap", "post", "CFF ", "hmtx", "DSIG")
+    # confirm that expected tables are printed to stdout
+    for table in expected_tables:
+        assert table in out
+    # test for one of the expected tag/checksum/length/offset strings
+    assert "OS/2  0x67230FF8        96       272" in out
+
+
+def test_ttx_ttlist_woff(capsys, tmpdir):
+    inpath = os.path.join("Tests", "ttx", "data", "TestWOFF.woff")
+    fakeoutpath = os.path.join(str(tmpdir), "TestWOFF.ttx")
+    options = ttx.Options([("", "")], 1)
+    options.listTables = True
+    options.flavor = "woff"
+    ttx.ttList(inpath, fakeoutpath, options)
+    out, err = capsys.readouterr()
+    expected_tables = ("head", "hhea", "maxp", "OS/2", "name", "cmap", "post", "CFF ", "hmtx", "DSIG")
+    # confirm that expected tables are printed to stdout
+    for table in expected_tables:
+        assert table in out
+    # test for one of the expected tag/checksum/length/offset strings
+    assert "OS/2  0x67230FF8        84       340" in out
+
+
+def test_ttx_ttlist_woff2(capsys, tmpdir):
+    inpath = os.path.join("Tests", "ttx", "data", "TestWOFF2.woff2")
+    fakeoutpath = os.path.join(str(tmpdir), "TestWOFF2.ttx")
+    options = ttx.Options([("", "")], 1)
+    options.listTables = True
+    options.flavor = "woff2"
+    ttx.ttList(inpath, fakeoutpath, options)
+    out, err = capsys.readouterr()
+    expected_tables = ("head", "hhea", "maxp", "OS/2", "name", "cmap", "hmtx", "fpgm", "prep", "cvt ", "loca", "glyf", "post", "gasp")
+    # confirm that expected tables are printed to stdout
+    for table in expected_tables:
+        assert table in out
+    # test for one of the expected tag/checksum/length/offset strings
+    assert "OS/2  0x67230FF8        96         0" in out
+
+# -------------------
+# main function tests
+# -------------------
+
+
+def test_ttx_main_default_ttf_dump_to_ttx(tmpdir):
+    inpath = os.path.join("Tests", "ttx", "data", "TestTTF.ttf")
+    outpath = os.path.join(str(tmpdir), "TestTTF.ttx")
```

Comment:
`tmpdir` fixture returns a py.path.local object; to join it with a sub-path, you can use the division operator, e.g. `tmpdir / "TestTTF.ttx"`.
See the full set of features here: http://py.readthedocs.io/en/latest/path.html

Reply:
It’s part of pytest, you’re already using it!

Reply:
oh... I didn't realize that.  ty :)

Reply:
@anthrotype 

I modified one of the functions to use the division operator for path joins with the pytest tmpdir fixture as follows:

```python
def test_ttcompile_otf_compile_default(tmpdir):
    inttx = os.path.join("Tests", "ttx", "data", "TestOTF.ttx")
    # outotf = os.path.join(str(tmpdir), "TestOTF.otf")
    outotf = tmpdir/"TestOTF.otf"
    default_options = ttx.Options([], 1)
    ttx.ttCompile(inttx, outotf, default_options)
    # confirm that font was built
    assert outotf.check(file=True)
    # confirm that it is valid OTF file, can instantiate a TTFont, has expected OpenType tables
    ttf = TTFont(outotf)
    expected_tables = (
        "head",
        "hhea",
        "maxp",
        "OS/2",
        "name",
        "cmap",
        "post",
        "CFF ",
        "hmtx",
        "DSIG",
    )
    for table in expected_tables:
        assert table in ttf
```

and I am getting this error raised from ttFont:

```
============================================= FAILURES =============================================
________________________________ test_ttcompile_otf_compile_default ________________________________

tmpdir = local('/private/var/folders/x7/vs8mbwmx1hg1vcb53nlygt0m0000gn/T/pytest-of-chris/pytest-97/test_ttcompile_otf_compile_def0')

    def test_ttcompile_otf_compile_default(tmpdir):
        inttx = os.path.join("Tests", "ttx", "data", "TestOTF.ttx")
        # outotf = os.path.join(str(tmpdir), "TestOTF.otf")
        outotf = tmpdir/"TestOTF.otf"
        default_options = ttx.Options([], 1)
>       ttx.ttCompile(inttx, outotf, default_options)

Tests/ttx/ttx_test.py:520:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
Lib/fontTools/misc/loggingTools.py:372: in wrapper
    return func(*args, **kwds)
Lib/fontTools/ttx.py:290: in ttCompile
    ttf.save(output)
Lib/fontTools/ttLib/ttFont.py:189: in save
    file.write(tmp2.getvalue())
/usr/local/lib/python3.6/site-packages/py/_path/local.py:500: in write
    data = py.builtin._totext(data, sys.getdefaultencoding())
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

obj = b'OTTO\x00\n\x00\x80\x00\x03\x00 CFF &\xfc\xc2\x10\x00\x00\x07\xf8\x00\x00\x00\xebDSIG\x00\x00\x00\x01\x00\x00\x08\xfc...x01\xf4\x002\x00\x00\x00\x00\x00\xfa\x00\x00\x00\xfa\x00\x00\x00\xf1\x007\x02\xd3\x007\x00\x00\x00\x01\x00\x00\x00\x00'
encoding = 'utf-8', errors = None

    def _totext(obj, encoding=None, errors=None):
        if isinstance(obj, bytes):
            if errors is None:
>               obj = obj.decode(encoding)
E               UnicodeDecodeError: 'utf-8' codec can't decode byte 0x80 in position 7: invalid start byte

/usr/local/lib/python3.6/site-packages/py/_builtin.py:125: UnicodeDecodeError
===================================== short test summary info ======================================
FAIL Tests/ttx/ttx_test.py::test_ttcompile_otf_compile_default
=============================== 1 failed, 78 passed in 2.01 seconds ================================
```

Thoughts?